tests/integration/024_cloud_credentials/main.tf

```diff
@@ -4,26 +4,35 @@ resource "random_string" "random" {
   min_lower = 8
 }
 
-resource "env0_aws_credentials" "cred1" {
+resource "env0_aws_credentials" "aws_cred1" {
   name = "Test Role arn1 ${random_string.random.result}"
   arn  = "Role ARN1"
 }
 
-resource "env0_gcp_credentials" "cred2" {
+resource "env0_aws_credentials" "aws_cred2" {
+  name = "Test Role arn2 ${random_string.random.result}"
+  arn  = "Role ARN2"
+}
+
+resource "env0_gcp_credentials" "gcp_cred" {
   name                = "name ${random_string.random.result}"
   service_account_key = "your_account_key"
   project_id          = "your_project_id"
 }
 
-data "env0_cloud_credentials" "aws_credentials" {
+data "env0_cloud_credentials" "all_aws_credentials" {
+  depends_on = [env0_aws_credentials.aws_cred1, env0_aws_credentials.aws_cred2, env0_gcp_credentials.gcp_cred]
   credential_type = "AWS_ASSUMED_ROLE_FOR_DEPLOYMENT"
 }
 
-data "env0_aws_credentials" "aws_credentials" {
-  for_each = toset(data.env0_cloud_credentials.aws_credentials.names)
-  name     = each.value
+data "env0_aws_credentials" "aws_credentials1" {
+  name     = data.env0_cloud_credentials.all_aws_credentials.names[index(data.env0_cloud_credentials.all_aws_credentials.names, env0_aws_credentials.aws_cred1.name)]
+}
+
+data "env0_aws_credentials" "aws_credentials2" {
+  name     = data.env0_cloud_credentials.all_aws_credentials.names[index(data.env0_cloud_credentials.all_aws_credentials.names, env0_aws_credentials.aws_cred2.name)]
```

Comment:
why was this part added? it's not used

Reply:
this verify you found the resource you created before,
otherwise it will fail

Reply:
`aws_credentials2` is not used. in the output we only output `1`, not even `GCP`, but I get why to create `GCP` too. what's the use of `aws2`?

Reply:
ok, got it, thanks